lib/events/s3_exporter.rb

```diff
@@ -0,0 +1,66 @@
+module Events
+  class S3Exporter
+    def initialize(created_before, created_on_or_after = nil)
+      @created_before = created_before
+      @created_on_or_after = created_on_or_after
+    end
+
+    def export
+      file = Tempfile.new("events_s3_#{filename}.csv.gz")
+      zipped = Zlib::GzipWriter.open(file) do |gzip|
+        gzip.orig_name = "#{filename}.csv"
+        build_csv(gzip)
+        gzip.close
+      end
+      upload(zipped)
+      exported = event_scope.update_all(payload: nil)
```

Comment:
I'll get a test in to cover that scenario, I've assumed an exception - but we all know what exceptions lead to...

yeah it could be set as an empty JSON object, queries for it look a bit gnarly http://stackoverflow.com/questions/24292575/how-to-query-a-json-column-for-empty-objects null feels a bit more fitting as it differentiates that from any ability someone may have to send us an empty payload.